lib/iterator.js

```diff
@@ -233,6 +233,16 @@ class Iterator {
     return map;
   }
 
+  // Consumes an iterator returning last iterator value
```

Comment:
Maybe
```suggestion
  // Consume an iterator and return last iterator value
```
since most comments use infinitive form?

Reply:
Current style is consistent with other such functions, we should change this for all of them if we want to.